### PR TITLE
#159952006 Users should be able to create articles

### PR DESCRIPTION
**Pull request by reiosantos**
#### What does this PR do?
- Enable a user to create articles and read/access articles created by other authors.
- Enable a user to ONLY update and delete articles he authored.

#### Description of Task to be completed?
- Ensure that a user is able to retrieve a list of all articles
- Ensure that a user is able to retrieve a specific article.
- Ensure that a user is able to update and delete articles he created

#### How should this be manually tested?
- use postman to make API calls.
-  Create `/api/articles/` method **`<POST>`**
```json
{
  "articles": {
     "title": "<article title>",
     "description": "<describe your story>",
     "body": "<story body>"
  }
}
```
- Retrieve all articles `/api/articles/` method **`GET`**
- Retrieve specific article `/api/articles/<article_id>/` method **`GET`**
- Delete article `/api/articles/<article_id>/` method **`DELETE`**
- Update article `/api/articles/` method **`PUT`** "`Like Create article`, howeve, The three fields are optional. You can miss out some of them."

#### What are the relevant pivotal tracker stories?
[#159952006](https://www.pivotaltracker.com/story/show/159952006)

### CODE REVIEW

**Comment by collinewait** [comment]
Missing docstrings(class docstrings and method docstrings)

**Comment by inno-asiimwe** [comment]
@reiosantos please resolve conflicts on this branch.

**Comment by reiosantos** [comment]
> @reiosantos please resolve conflicts on this branch.

All conflicts have been resolved

**Comment by reiosantos** [comment]
> Missing docstrings(class docstrings and method docstrings)

I have added the docstrings for all methods/classes

File: authors/apps/authentication/models.py [hunk 1]
```diff
@@ -145,3 +144,15 @@ def generate_token(self):
         }, settings.SECRET_KEY, algorithm='HS256')
 
         return token.decode('utf-8')
+
```
Comment (inno-asiimwe) [inline authors/apps/authentication/models.py:147]: Remove all commented out code @reiosantos.

**Pull request merged**